velox/common/memory/tests/MemoryUsageTrackerTest.cpp

```diff
@@ -315,3 +315,34 @@ TEST(MemoryUsageTrackerTest, maybeReserve) {
   // The child destruction won't release the reserved memory back to the parent.
   EXPECT_EQ(8 * kMB, parent->getCurrentUserBytes());
 }
+
+TEST(SimpleMemoryUsageTracker, testHierarchicalUpdate) {
+  auto root = SimpleMemoryTracker::create();
```

Comment:
It would be great if you could write a concurrent tests for this like:
1. start a bunch of threads.
2. for each thread (say 20), in a loop: either one of (1) pickup an existing memory pool (shared pointer referenced) do memory allocation, and free. And do optional deletes; (2) create a new child, and memory allocation and free;
you can put a limit on the number of pools and run the test for 5 seconds say and they set a stop signal to stop all the threads.
You can put the created memory pool in unordered_set or ordered_map witb index by the memory pool name which could be monotonically increasing counter string (std::to_string(nextMemoryPoolId))